pkg/util/azureclient/roundtripper.go

```diff
@@ -0,0 +1,84 @@
+package azureclient
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	utillog "github.com/Azure/ARO-RP/pkg/util/log"
+)
+
+const (
+	genevaOutboundRequestsTable = "outboundRequests"
+)
+
+func NewCustomRoundTripper(next http.RoundTripper) http.RoundTripper {
+	return &customRoundTripper{
+		next: next,
+	}
+}
+
+type customRoundTripper struct {
+	next http.RoundTripper
+}
+
+func (crt *customRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	startTime := time.Now()
+
+	correlationData := api.GetCorrelationDataFromCtx(req.Context())
+	if correlationData == nil {
+		correlationData = api.CreateCorrelationDataFromReq(req)
+	}
+
+	l := updateCorrelationDataAndEnrichLog(correlationData, utillog.GetLogger(), startTime)
+	l.Info("HttpRequestStart")
+
+	res, err := crt.next.RoundTrip(req)
+
+	l = updateCorrelationDataAndEnrichLogWithResponse(correlationData, l, res, startTime)
+	l.Info("HttpRequestEnd")
+
+	return res, err
+}
+
+// updateCorrelationDataAndEnrichLog receives a non nil correlationData and updates the start time. It also returns a new logrus entry
+// updated with the new field for the LOGKIND field, corresponding to a specific custom DGrep table for outgoing requests.
+func updateCorrelationDataAndEnrichLog(correlationData *api.CorrelationData, l *logrus.Entry, startTime time.Time) *logrus.Entry {
```

Comment:
Is this type of docstring comment a convention we follow?

Reply:
Comments in Go have a special convention followed by the community: "Go Doc Comments". We don't always use them in the project but I find them useful sometimes to explain functions that are not trivial. Ideally you want to document the public API of your packages, but we don't do that strictly->this is not the case, just a general comment -never better said : ) 

In that case I think the comment makes sense because:
- it explicitly says that correlation data should not be nil (explicit precondition)
- it adds context about what is the LOGKIND field as it is a trivial convention that we use to configure our logging to write to a different table (this is also explained in the Spike as it is not trivial)

Here is the explanation of the standard convention: https://tip.golang.org/doc/comment

WDYT ?

Reply:
That's perfect. Ty for the context

Reply:
No problem. Sometimes the line between useful and redundant in that kind of comments is not 100% clear.